app/routes/web/benchmark.py

```diff
@@ -58,15 +87,23 @@ def benchmark(
         app.session.logger.warning(f'Failed to submit benchmark: Restricted')
         raise HTTPException(401)
 
-    users.update(player.id, {'latest_activity': datetime.now()}, session)
+    try:
+        hardware_dict = json.loads(hardware)
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=400, detail="Invalid hardware format")
+
+    validate_hardware_data(hardware_dict)
```

Comment:
I would suggest doing the json decoding inside the `validate_hardware_data` function instead.
For example:
```python
hardware_dict = validate_hardware_data(hardware)
```

Reply:
I have moved json decoding to the validate_hardware_data function